GitUI/GitUIExtensions.cs

```diff
@@ -198,6 +198,49 @@ private static IEnumerable<TreeNode> AllNodes(this TreeNodeCollection nodes)
             }
         }
 
+        /// <summary>
+        /// Enumerates all descendant controls.
+        /// </summary>
+        public static IEnumerable<Control> FindDescendants([NotNull] this Control control)
+        {
+            var queue = new Queue<Control>();
+
+            foreach (Control child in control.Controls)
+            {
+                queue.Enqueue(child);
+            }
+
+            while (queue.Count != 0)
+            {
+                var c = queue.Dequeue();
+
+                yield return c;
+
+                foreach (Control child in c.Controls)
+                {
+                    queue.Enqueue(child);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Enumerates all descendant controls of type <typeparamref name="T"/> in breadth-first order.
+        /// </summary>
+        public static IEnumerable<T> FindDescendantsOfType<T>([NotNull] this Control control) where T : Control
+        {
+            return FindDescendants(control).OfType<T>();
+        }
+
+        /// <summary>
+        /// Finds the first descendent of <paramref name="control"/> that has type
+        /// <typeparamref name="T"/> and satisfies <paramref name="predicate"/>.
+        /// </summary>
+        [CanBeNull]
+        public static T FindDescendantsOfType<T>(this Control control, Func<T, bool> predicate) where T : Control
```

Comment:
❗️ This should be singular.

:bulb: In addition making the method singular, consider using a name that indicates `default(T)` could be returned (e.g. `*OrDefault` or `Try*`). Alternately, you could just remove the use of `FirstOrDefault` and consolidate this with the previous method (making `predicate` optional and act as `x=>true` when omitted).

Reply:
Thanks Sam.
Holding off until Drew merges his PR in, then will address your comments.

Reply:
The pluralisation of that method name was fixed in #5087.